bin/test-run

```diff
@@ -31,8 +31,36 @@ function check_if_k8s_reachable(){
 }
 
 function run_test(){
-    printf "Running test [%s] %s\\n" "$(basename "$1")" "$2"
-    go test -v "$1" --linkerd="$linkerd_path" --linkerd-namespace="$linkerd_namespace" --k8s-context="$k8s_context" --integration-tests "$2"
+    filename="$1"
+    shift
+
+    printf "Running test [%s] %s\\n" "$(basename "$filename")" "$@"
+    go test -v "$filename" --linkerd="$linkerd_path" --k8s-context="$k8s_context" --integration-tests "$@"
```

Comment:
Replaced the `"$2"` with `"$@"`, because the former converts multiple arguments into one long, space-delimited argument. For example, when running `run_test filename --linkerd-namespace=linkerd --proxy-auto-inject`, the `testutil.TestHelper` ends up treating the linkerd namespace as `linkerd --proxy-auto-inject`.

Reply:
Also, removed. the `--linkerd-namespace="$linkerd_namespace"` argument. This should be part of `$@`, instead of relying on the global `$linkerd_namespace` variable. This was problematic for upgrade and auto-inject test cases which depend on the control plane in a different namespace.